### PR TITLE
docs: fabric-peer-ext documentation

### PR DESCRIPTION
**Pull request by talwinder50**
This is initial documentation skeleton for this project 

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>
#114

### CODE REVIEW

**Comment by fqutishat** [comment]
verify-unitest

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Minimal makefile for Sphinx documentation
+#
```
Comment (troyronda) [inline docs/Makefile]: Should include SPDX header in all files.

**Comment by talwinder50** [comment]
verify-bddtest

File: docs/source/roles.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+TODO additional comments
+
+These roles are:
+
+Endorser
+--------
+
+An endorser peer will only do endorsements of transactions and will never
+right to the ledger, it will update its data from ledger periodically through
```
Comment (rolsonquadras) [inline docs/source/roles.rst]: I believe it should be 'write', instead of 'right'.

File: docs/source/roles.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+TODO additional comments
+
+Committer
+---------
+
+A committer peer will deal with only committing transactions to the ledger.
+This way all other peers are relieved from writing to the ledger and execute
+a larger number of endorsements.
+
+There should be only 1 endorser for a given org.
```
Comment (rolsonquadras) [inline docs/source/roles.rst]: This should be committer and not endorser.

File: docs/source/blkstorage.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
+Block Storage Enhancements
+==========================
+
+Block Storage was originally implemented with a FileSystem storage using
+indexes stored in LevelDB. This filesystem storage is not scallable and
```
Comment (rolsonquadras) [inline docs/source/blkstorage.rst]: typo - scalable

**Comment by talwinder50** [comment]
verify-unitest

File: docs/source/roles.rst (hunk 3 above)
Comment (talwinder50) [reply docs/source/roles.rst]: edited

File: docs/source/blkstorage.rst (hunk 4 above)
Comment (talwinder50) [reply docs/source/blkstorage.rst]: edited

File: docs/source/collections.rst [hunk 5]
```diff
@@ -0,0 +1,12 @@
+Private Collections Enhancements
+================================
+
+Private Collections are not supported a Distributed Content Addressable Storage type (or DCAS)
```
Comment (rolsonquadras) [inline docs/source/collections.rst]: May be we need to re-word this sentence. It's confusing.

File: docs/source/roles.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
+Peer Roles Enhancements
+=======================
+
+The peers are not split into two roles to enhance endorsements and commit
```
Comment (rolsonquadras) [inline docs/source/roles.rst]: should it be now ?

File: docs/source/pvtdatastorage.rst [hunk 7]
```diff
@@ -0,0 +1,16 @@
+Private Data Storage Enhancements
+=================================
+
+Private Data Storage was also enhanced with CouchDB
+
+Enhancement 1
```
Comment (rolsonquadras) [inline docs/source/pvtdatastorage.rst]: can we name it better than just saying "Enhancement 1" ?

File: docs/source/collections.rst (hunk 5 above)
Comment (talwinder50) [reply docs/source/collections.rst]: done

File: docs/source/pvtdatastorage.rst (hunk 7 above)
Comment (talwinder50) [reply docs/source/pvtdatastorage.rst]: done

File: docs/source/roles.rst (hunk 6 above)
Comment (talwinder50) [reply docs/source/roles.rst]: done

File: docs/source/blkstorage.rst [hunk 8]
```diff
@@ -0,0 +1,18 @@
+Block Storage Enhancements
+==========================
+
+Block Storage was originally implemented with a FileSystem storage using
+indexes stored in LevelDB. This filesystem storage is not scalable and
+requires a separate storage for the indexes. The new storage used in this
+mod project is CouchDB
```
Comment (rolsonquadras) [inline docs/source/blkstorage.rst]: period at the end of the sentence.

Comment (talwinder50) [reply docs/source/blkstorage.rst]: Added

**Pull request merged**